common/model-views.cpp

```diff
@@ -5764,14 +5764,23 @@ namespace rs2
                     ///////////////////////////////////////////
                     //TODO: make this a member function
                     std::vector<const char*> labels;
+                    std::vector< int > counters;
                     auto selected = 0, counter = 0;
-                    for (auto i = opt_model.range.min; i <= opt_model.range.max; i += opt_model.range.step, counter++)
+                    // After changing preset the default values of some controls may change
+                    // So we need to refrash the cached values
+                    opt_model.update_all_fields( error_message, model );
+                    for (auto i = opt_model.range.min; i <= opt_model.range.max; i += opt_model.range.step)
                     {
+                        if (i == RS2_L500_VISUAL_PRESET_DEFAULT)
+                            continue;
```

Comment:
`// Default is only there for backwards compatibility and will throw an exception if used`

Also, what about automatic? Does it not need to be ignored, too?

Reply:
Add the comment

Reply:
done